### DISCUSSION TITLE
Agama Users (in Unattended mode): Only root and the first user are supported

### DISCUSSION THREAD

**Discussion by kobliha** [agama-project/agama#1281] [Ideas]
I understand that this is still an open question: **How much should Agama be backward-compatible.** Anyway, the point is that more users are definitely being supported and used by customers in AutoYaST and we should try hard to get more data / decision about the to-be supported scenario(s).

The answer, on the other hand, should not be based on, e.g., having the functionality currently written in Perl. If it's needed, it's needed and we have to find a way. The issue exists as a topic-opener and it's mean for gathering feedback too.

**Comment by imobachgs** [comment]
If you do not mind, I will move this issue to a discussion. We can create issues when we know what we decide what we are going to implement.

**Comment by imobachgs** [comment]
I have the same question, especially regarding users. In general, I do not like how users are handled in AutoYaST. Most of the users you can find on a typical system are already created by their own packages. So, once you install the same packages, you do not need to create them at all. The problem is that if you clone a system, you get the full list of users/groups. For that reason, we introduced the "compact" mode in the past, so only non-system users are cloned.

But is creating a single user (first user) and the root authentication mechanism enough? Well, when it comes to the ideas behind Agama, it should be. Any extra stuff should be handled after the first boot (out of Agama's territory). But I can understand that if you are using an AutoYaST profile, things might be different.

Regarding this topic, I have some questions:

* Do we want to open the door to enable different features, depending on whether you are doing an unattended or manual installation? Well, we are already doing that by supporting scripts.
* Should we extend the UI to support creating multiple users?
* Or should we support this feature only in AutoYaST mode?

OK, more questions and no answers, but the topic is worth a discussion :wink:

**Comment by kobliha** [comment]
I have another question: What about software? AutoYaST supports selecting particular packages, but Agama supports only patterns. If we support 3rd party add-ons/repos, then we may need to support packages (at least) in unattended more too. And those repositories (URLs) actually as well.

**Comment by rreed9991** [comment]
Is there any documentation showing how to create a basic unattended iso with SLES 16? I have the GM release downloaded yet there is no documentation that I can find showing how to do this.

**Reply by imobachgs** [reply]
If you do not know how unattended installation works in Agama, I would recommend you to have a look to our starting guide (especially to the unattended installation section): https://agama-project.github.io/docs/overview. However, you will find official documentation once SLE 16 is released.

**Reply by rreed9991** [reply]
We are currently unable to complete an unattended installation of the SLE 16 GM release using any of the existing Agama configuration examples, along with several tweaks I made to match the product ID. The unattended installation stops at the Product Selection page, regardless of the product ID I gave it.
Due to the lack of SLE 16.0 GA documentation, could you provide a basic Agama profile example (Jsonnet or JSON) that is more likely to resolve the product identification issue that I am seeing? It’s okay if it doesn’t work, but leveraging your expertise would be helpful.

**Reply by rreed9991** [reply]
If this helps at all:

isoinfo -d -i SLES-16.0-Full-x86_64-GM.install.iso
Setting input-charset to 'UTF-8' from locale.
CD-ROM is in ISO 9660 format
System id: LINUX
Volume id: Install-SUSE-SLE-16-x86_64
Volume set id:
Publisher id: SUSE LLC
Data preparer id: Product Composer - http://www.github.com/openSUSE/product-composer
Application id: SLES-16.0-Full-x86_64-Build135.5
Copyright File id:
Abstract File id:
Bibliographic File id:
Volume set size is: 1
Volume set sequence number is: 1
Logical block size is: 2048
Volume size is: 4925345
El Torito VD version 1 found, boot catalog is in sector 20

Joliet with UCS level 3 found.
SUSP signatures version 1 found
Rock Ridge signatures version 1 found
Rock Ridge id 'RRIP_1991A'
Eltorito validation header:
    Hid 1
    Arch 0 (x86)
    ID 'SUSE LLC'
    Cksum 98 2B OK
    Key 55 AA
    Eltorito defaultboot header:
        Bootid 88 (bootable)
        Boot media 0 (No Emulation Boot)
        Load segment 0
        Sys type 0
        Nsect 4
        Bootoff 29FA 10746

**Reply by imobachgs** [reply]
Please, I would need more information about what you are trying and the logs, as usual. It works for me with the following configuration:

```jsonnet
{
  "user": {
    "fullName": "SUSE",
    "userName": "suse",
    "password": "my-password",
  },
  "product": {
    "id": "SLES",
  }
}
```

However, this discussion was about "Only root and the first user are supported" so, please, open a new discussion so we can continue there.

**Comment by imobachgs** [comment]
This discussion is rather old. JFTR, we are keeping the same behavior regarding users, but in packages can be selected individually using an Agama configuration file (not in the UI).
